8/SiloHost/src/Program.cs

```diff
@@ -9,70 +8,29 @@
 using Orleans.Hosting;
 using Orleans.Statistics;
-using System;
-using System.Net;
-using System.Net.NetworkInformation;
-using System.Net.Sockets;
 using System.Threading.Tasks;
+using Amazon.Util.Internal;
 
 namespace SiloHost
 {
     class Program
     {
+        private static readonly EnvironmentVariables environmentVariable = new EnvironmentVariables();
+        private readonly string EcsContainerMetadataUri = environmentVariable.EcsContainerMetadataUri();
+
         public static Task Main()
         {
-            var advertisedIp = Environment.GetEnvironmentVariable("ADVERTISEDIP");
-            var advertisedIpAddress = advertisedIp == null ? GetLocalIpAddress() : IPAddress.Parse(advertisedIp);
-
-            var extractedGatewayPort = Environment.GetEnvironmentVariable("GATEWAYPORT")?? throw new Exception("Gateway port cannot be null");
-            var extractedSiloPort = Environment.GetEnvironmentVariable("SILOPORT")?? throw new Exception("Silo port cannot be null");
-            var extractDashboardPort = Environment.GetEnvironmentVariable("DASHBOARDPORT") ?? throw new Exception("Dashboard port cannot be null");
-            var extractedPrimaryPort = Environment.GetEnvironmentVariable("PRIMARYPORT") ?? throw new Exception("Primary port cannot be null");
-            var awsRegion = Environment.GetEnvironmentVariable("AWSREGION") ?? throw new Exception("Aws region cannot be null");
-            var membershipTable =  Environment.GetEnvironmentVariable("MEMBERSHIPTABLE") ?? throw new Exception("Membership table cannot be null");
-            var primaryAddress = Environment.GetEnvironmentVariable("PRIMARYADDRESS") ?? throw new Exception("Primary address cannot be null");
-
-            var siloPort = int.Parse(extractedSiloPort);
-            var developmentPeerPort = int.Parse(extractedPrimaryPort);
-            var gatewayPort = int.Parse(extractedGatewayPort);
-            var dashboardPort = int.Parse(extractDashboardPort);
-            var primaryIp = IPAddress.Parse(primaryAddress);
-
-            var primarySiloEndpoint = new IPEndPoint(primaryIp, developmentPeerPort);
-
-            var siloEndpointConfiguration = new SiloEndpointConfiguration(advertisedIpAddress, siloPort, gatewayPort);
+            ISiloConfigurationHelper siloConfigurationHelper = new SiloConfigurationHelper(environmentVariable);
 
             return new HostBuilder()
                 .UseOrleans(siloBuilder =>
                 {
                     siloBuilder.UseLinuxEnvironmentStatistics();
-                    siloBuilder.UseDashboard(dashboardOptions =>
-                    {
-                        dashboardOptions.Username = "piotr";
-                        dashboardOptions.Password = "orleans";
-                        dashboardOptions.Port = dashboardPort;
-                    });
-
+                    siloBuilder.UseDashboard(dashboardOptions => siloConfigurationHelper.ConfigureDashboardOptions(dashboardOptions));
```

Comment:
Let's consider writing our own `UseDashboardXYZ` extension - it would look much nicer.